### ISSUE TITLE
brace_linter() should still lint when allow_single_line=TRUE but { is on its own line

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#1684]
The following tests currently pass; I think they shouldn't. `allow_single_line` is meant to be about `{` matching `}` on the same line, not about whether `{` / `}` are on their own line

```r
expect_lint("while (FALSE)\n{}", NULL, brace_linter(allow_single_line = TRUE))
expect_lint("while (FALSE)\n{ }", NULL, brace_linter(allow_single_line = TRUE))
```
